File: app/src/main/kotlin/org/triplepy/sh8email/sh8/activities/login/di/LoginModule.java
package org.triplepy.sh8email.sh8.activities.login.di;


/**
 * The sh8email-android Project.
 * ==============================
 * org.triplepy.sh8email.sh8.activities.login.di
 * ==============================
 * Created by igangsan on 2016. 9. 3..
 */

import org.triplepy.sh8email.sh8.api.ClientModule;
import org.triplepy.sh8email.sh8.activities.login.presenter.LoginPresenter;

import dagger.Module;
import dagger.Provides;

@Module(includes = ClientModule.class)
public class LoginModule {
    private LoginPresenter.View view;

    public LoginModule(LoginPresenter.View view){
        this.view = view;
    }

    @Provides
    public LoginPresenter.View provideLoginView(){
        return view;
    }
}

File: app/src/main/kotlin/org/triplepy/sh8email/sh8/activities/login/di/LoginModule.kt
package org.triplepy.sh8email.sh8.activities.login.di

import dagger.Module
import dagger.Provides
import org.triplepy.sh8email.sh8.activities.login.presenter.LoginPresenter
import org.triplepy.sh8email.sh8.activities.login.presenter.LoginPresenterImpl
import org.triplepy.sh8email.sh8.di.module.ClientModule

/**
 * The sh8email-android Project.
 * ==============================
 * org.triplepy.sh8email.sh8.activities.login.di
 * ==============================
 * Created by igangsan on 2016. 9. 3..
 */
@Module(includes = arrayOf(ClientModule::class))
class LoginModule {
    val view: LoginPresenter.View

    constructor(view: LoginPresenter.View) {
        this.view = view
    }

    @Provides
    fun provideLoginPresenter(loginPresenter: LoginPresenterImpl): LoginPresenter {
        return loginPresenter
    }

    @Provides
    fun provideLoginView(): LoginPresenter.View {
        return view
    }
}